### DISCUSSION TITLE
Failed to fetch Snap "local:http://localhost:8080": Failed to fetch.

### DISCUSSION THREAD

**Discussion by PTS3000** [MetaMask/snaps#1719] [Q&A] [answered]
Hey! Just starting out on Metamask Snaps, following the tutorial. Stopped at the "Connect" step. Basically, I get the error:

Failed to fetch Snap "local:http://localhost:8080": Failed to fetch. Every time I click the "Connect" button. My snap.manifest.json and package.json are located in the server root directory. So I don't think the problem lies there. Any ideas on what I could do?

**Comment by promet99** [comment]
Try using node 18

**Reply by PTS3000** [reply]
How do I change that parameter?

**Reply by promet99** [reply]
`nvm use 18`

**Reply by PTS3000** [reply]
Hey there. I tried using node 18.2.0 but I still get the exact same error.

**Comment by Montoya** [comment]
Hi, have you tried using the template snap monorepo? https://github.com/MetaMask/template-snap-monorepo

Try cloning that locally, then do `yarn && yarn start` and see if the snap and dapp are both able to run simultaneously. If so, you will be able to open the dapp and connect to the snap.

**Reply by PTS3000** [reply]
Hey @Montoya. So everything works up until running the snap locally. I'm using node js 18.2.0. I cloned the snap monorepo. But I still get the same mistake. Is there something wrong with my connection request? (print attached)
<img width="435" alt="Captura de ecrã 2023-09-07, às 10 46 50" src="https://github.com/MetaMask/snaps/assets/62572822/c3fb7b0c-e5af-463d-a566-24be337e530f">

**Reply by PTS3000** [reply]
The error is now: Failed to fetch Snap "local:http://localhost:8080": At path: version -- Expected a string, but received: undefined.

**Comment by PTS3000** [comment]
So apparently the problem was I was initiating the snap in the wrong port. Now I get a different error:
Failed to fetch Snap "local:http://localhost:8080": At path: version -- Expected a string, but received: undefined.
My package.json and snap.manifest.json have all the version fields filled though. 

Any thoughts?

**Reply by Montoya** [reply, marked as answer]
Are you sure that both the dapp and the snap are running? If you run `yarn install && yarn start` in the template snap monorepo main directory, it will run both packages. The [site] will be running at :8000 and the [snap] will be running at :8080. That is necessary for one to connect to the other.

**Reply by PTS3000** [reply]
Hey! It appears to be working correctly now! Thank you :)
